core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -1276,69 +1281,145 @@ private static void mergeRowGroup(
             RowGroupBuffers rowGroupBuffers,
             int rowGroupIndex,
             int timestampIndex,
-            int timestampColumnType,
+            long partitionTimestamp,
             long mergeRangeLo,
             long mergeRangeHi,
             TableRecordMetadata tableWriterMetadata,
-            long srcOooBatchRowSize
+            long srcOooBatchRowSize,
+            long dedupColSinkAddr
     ) {
+        // decode column chunks for the row group in advance
         columnsIdsAndTypes.clear();
-        columnsIdsAndTypes.add(timestampIndex);
-        columnsIdsAndTypes.add(timestampColumnType);
+        int timestampColumnChunkIndex = -1;
+        final int columnCount = tableWriterMetadata.getColumnCount();
+        for (int column = 0; column < columnCount; column++) {
```

Comment:
Is it guaranteed that the table writer order and the parquet column order matches? Considering deleted columns that we skip (`columnType < 0`), this is questionable.

Reply:
A todo has been added. Can be a problem after adding the DDL.